### PR TITLE
Add pre-commit hook using husky for linting and formatting

### PR DESCRIPTION
**Pull request by choidabom**
#### What this PR does / why we need it?

This PR adds a pre-commit hook script using Husky to automatically lint and format code changes in the frontend and backend directories before each commit. The commit will be aborted if any linting or formatting errors are detected, which helps maintain code consistency and quality throughout the codebase.

#### Any background context you want to provide?

Previously, the codebase did not have a pre-commit hook to enforce linting and formatting checks before commit, leading to inconsistencies and potential errors in the code. 

- With this implementation, Husky has been configured to work in both the frontend and backend directories. 
- The pre-commit hook now detects changes in each directory separately and runs linting and formatting checks only on the modified directories. 
- If any linting errors are detected, the commit will be aborted.

#### What are the relevant tickets?

Fixes https://github.com/yorkie-team/codepair/issues/276

### Checklist
- [x] Added relevant tests or not required
- [x] Didn't break anything

## Summary by CodeRabbit

- **New Features**
	- Introduced automatic linting and formatting checks before commits for both frontend and backend code.
	- Added scripts to set up Git hooks using Husky, enhancing code quality assurance during development.

- **Chores**
	- Updated package.json files in both frontend and backend to include Husky as a dependency and added necessary setup scripts.

### CODE REVIEW

**Review by devleejb — CHANGES_REQUESTED** [review]
I have a question.

I remember the context we discussed on Discord last time. However, if we're going to set up `pre-commit`, wouldn't it be better to use `lint-staged`?

This way, we can limit the checks to only staged files, and Lint checks won't be performed on simple document edits like `md` files.

@gwbaik9717 How do you think about this?

**Comment by gwbaik9717** [comment]
@devleejb @choidabom 

Absolutely! There doesn't seem to be any downside or side-effects to adding `lint-staged`, since we've already decided to use `pre-commit`.

**Comment by choidabom** [comment]
@devleejb @gwbaik9717 
I agree! Then I will apply `lint-staged` to both the backend and frontend package.json.

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    LINT_BACKEND=true
+  fi
+done
+
+if [ "$LINT_FRONTEND" = true ]; then
+  echo "Changes detected in the frontend. Linting & Formatting frontend..."
+  cd frontend
+  npx lint-staged
+  if [ $? -ne 0 ]; then
+    echo "Frontend linting failed. Commit aborted."
```
Comment (devleejb) [inline .husky/pre-commit:19]: It should be `Frontend Linting & Formatting failed. Commit aborted.`.

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  fi
+  cd ..
+fi
+
+if [ "$LINT_BACKEND" = true ]; then
+ echo "Changes detected in the backend. Linting & Formatting backend..."
+  cd backend
+  npx lint-staged
+  if [ $? -ne 0 ]; then
+    echo "Backend linting failed. Commit aborted."
```
Comment (devleejb) [inline .husky/pre-commit:30]: It should be `Backend Linting & Formatting failed. Commit aborted.`.

**Review by devleejb — CHANGES_REQUESTED** [review]

**Review by devleejb — APPROVED** [review]
Thank you for your contribution.

**Pull request merged**